src/Chapter2.hs

```diff
@@ -858,8 +870,13 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
-
+rewind :: [a] -> [a]
+rewind l = go l []
+    where
+        go [] [] = []
+        go [] (x:xs) = (x:xs)
+        go (y:ys) [] = go ys [y]
+        go (y:ys) (x:xs) = go ys (y:x:xs)
```

Comment:
These two also can be combined:

```haskell
go (y:ys) acc = go ys (y:acc)
```

Reply:
Thank you. I really like the "recursive go" pattern but I need some more practice in the pattern-matching syntax to make the best use of it. You guys cover a lot of neat stuff in this project.